addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -243,28 +256,31 @@ function factory(dependencies) {
             if (!this.fileType) {
                 return false;
             }
-            return this.fileType.includes('text');
+            return this.fileType === 'text';
         }
 
         /**
          * @private
          * @returns {boolean}
          */
         _computeIsViewable() {
-            return (
-                this.mediaType === 'image' ||
-                this.mediaType === 'video' ||
-                this.mimetype === 'application/pdf' ||
-                this.isTextFile
-            );
-        }
-
-        /**
-         * @private
-         * @returns {string}
-         */
-        _computeMediaType() {
-            return this.mimetype && this.mimetype.split('/').shift();
+            switch (this.mimetype) {
+                case 'application/pdf':
+                case 'image/bmp':
+                case 'image/gif':
+                case 'image/jpeg':
+                case 'image/png':
+                case 'image/webp':
+                case 'text/css':
+                case 'text/html':
+                case 'text/plain':
+                case 'video/mp4':
+                case 'video/ogg':
+                case 'video/webm':
+                    return true;
+                default:
+                    return false;
+            }
```

Comment:
All the url attachments become not viewable then? Wasn't it possible to view url image before?

Reply:
Before PR, `isViewable` relied on `mediaType` for image & video.
`mediaType` was computed using `mimetype`'s 1st part.
If some URL attachments were viewable before but no longer after this PR, this would mean they have `mimetype` value `image/X` or `video/X`. I think it's fine not making them viewable unless we list them explicitly.